### ISSUE TITLE
[Feature Request] Increase Storage Capacity in Mithril Chests

### ISSUE THREAD

**Issue by DustyBagel** [minetest-mods/technic#626]
I have noticed that the gold chest and the mithril chest have the same amount of item slots. This doesn't make sense because of the resources needed to make a mithril chest.

**Comment by SmallJoker** [comment]
Copper chest: 12 * 5 = 60 slots
Silver chest: 12 * 6 = 72 slots
Gold chest: 15 * 6 = 90 slots

How many slots should the mithril chest have? You can experiment a bit with the inventory dimensions here:
https://github.com/minetest-mods/technic/blob/b221d697179ab60026a60d5a3e44972806b26e12/technic_chests/mithril_chest.lua#L30-L48

Make sure that you remove the chest and place it again after the change so that the inventory dimensions are initialized as intended.

Follow-up question: how should the chest be presented to mobile players? The mithril chest is already very tiny to work with. There are the following options: (unless you have more ideas)
 * Scrollbar container (entirely client-side)
 * Buttons to switch between pages (server delay due to formspec update)

**Comment by DustyBagel** [comment]
Well, if formspec size is a concern then one possible solution would be to make the rest of the chests a bit smaller. I will play around with the chest sizes and get back to you. How would you like the changes I make to be submitted? Via pull request or just tell you the sizes as making an entire pull request seems to be unnecessary.

**Comment by SmallJoker** [comment]
@DustyBagel I don't have a preference in that regard. Both would allow discussions with others to figure out the optimum balance between the chest sizes.

**Comment by cx384** [comment]
I agree that the resource cost of mithril chest makes no sense, but just giving them more slots is a bit uninspiring, since mithril is a magical metal.

Imo it would be much better if the chest could do something special. For example:

- More items per slot (using `count_meta`), or hiding slots that have the same item
- Automatic compressing (crafting 9 ingots onto blocks)
- Configurable injection functionality
- Charging/repairing items if placed next to a battery box/tool workshop
- Automatically taking selected items from the player inventory on punching it
- ...

**Comment by DustyBagel** [comment]
@cx384 Wait, mithril is a magical metal? I thought it was just fictional.

**Comment by cx384** [comment]
OK, calling it a magical metal may be a big stretch, but at least I wouldn't say it's non-magical.
In Tolkien's work it was not explicitly magical, I think.
However, nowadays it's a widespread material in the fantasy genre, where it sometimes has magical properties.

Regrades of whether mithril is magical, technic chests gain abilities when you upgrade them (`autosort`, `infotext`, `color` ...), but when you upgrade from gold to mithril you lose the `infotext` and `color` ability.
So I would like get a new ability instead.

**Comment by DustyBagel** [comment]
Well, we could do what minecraft did with the ender chest but that is an unoriginal idea that other mods have added anyway. Maybe sense mithril is supposed to be really strong, we could make the chest TNT proof.

**Comment by OgelGames** [comment]
Mithril chests in [`technic_plus`](https://github.com/mt-mods/technic) have digilines functionality :)

**Comment by DustyBagel** [comment]
I like the idea of making the stack sizes in the chest bigger. Digilines functionality is a good idea too. What does that do exactaly. I never figured out how digilines works.

**Comment by DustyBagel** [comment]
So, what is happening with this? Are you guys waiting on me to make the changes and create a pull request or is someone else handling it? I am all for giving the mithril chest some kind of special ability, but I have no idea what that should be and I have been pretty busy with other projects so I would appreciate it if someone would please fill me in.

**Comment by cx384** [comment]
I think this is still in discussion, I would opt for at least digilines functionality.

So far, there wasn't much opposition to any of the proposed features, so you could make a pull request if you have a concrete implementation that you think will be accepted.

**Comment by Desour** [comment]
More ideas:

You could also make the inventory one slot bigger for each stack with stackcount 1 that is in the inventory, up until a max size is reached.

Or make it so that if you open the chest with a mtg key to a different mithril chest, you open the inventory of that other chest. (Maybe eat up the key for less cheatiness.)
